### ISSUE TITLE
on Windows 10, TimeoutExpired exception gets TypeError

### ISSUE THREAD

**Issue by DavidJea** [google/python-subprocess32#7]
Run into an issue, see below, when using check_output and timeout.  This is on Windows 10, and understand that "This code has not been tested on Windows or other non-POSIX platforms.", but thought you might want to know.

```
C:\Users\djea>python -V
Python 2.7.10

C:\Users\djea>pip show subprocess32

---
Metadata-Version: 1.1
Name: subprocess32
Version: 3.2.7
Summary: A backport of the subprocess module from Python 3.2/3.3 for use on 2.x.
Home-page: https://github.com/google/python-subprocess32
Author: Gregory P. Smith
Author-email: greg@krypto.org
License: PSF license
Location: c:\python27\lib\site-packages
Requires:
```

When running this code,

```
import subprocess32
try:
    p = subprocess32.check_output("timeout 5",shell=True,timeout=3)
    print p
except subprocess32.TimeoutExpired:
    print "expired"
```

Will get this Error

```
Traceback (most recent call last):
  File "C:\Users\djea\timeout.py", line 5, in <module>
    p = subprocess32.check_output("timeout 5",shell=True,timeout=3)
  File "C:\Python27\lib\site-packages\subprocess32.py", line 631, in check_output
    output, unused_err = process.communicate(timeout=timeout)
  File "C:\Python27\lib\site-packages\subprocess32.py", line 927, in communicate
    stdout, stderr = self._communicate(input, endtime, timeout)
  File "C:\Python27\lib\site-packages\subprocess32.py", line 1190, in _communicate
    raise TimeoutExpired(self.args)
TypeError: __init__() takes at least 3 arguments (2 given)
```

Same code (except using 'sleep' instead of 'timeout') has no problem on Linux.

**Comment by nirs** [comment]
TimeoutExpired(*self.args) ?

**Comment by theyoprst** [comment]
@gpshead any plans to fix it? :)

**Comment by gpshead** [comment]
Not high on my list, people shouldn't use subprocess32 on Windows. It is only meant for POSIX systems.

A pull request from a windows user would help.

**Comment by nirs** [comment]
#8 is a possible fix, not sure this is the fix Windows users would like :-)

**Comment by codeskyblue** [comment]
meet the same problem too

**Comment by samkaufman** [comment]
I am running into this issue on Linux. Is there a reason to think this error is Windows-specific?

**Comment by gpshead** [comment]
I believe https://github.com/google/python-subprocess32/pull/32 addressed this?  included in 3.5.0rc1.

**Comment by strayboy** [comment]
in centOS, the same error also, appear,here is :+1: 
`Exception in thread Thread-3:
Traceback (most recent call last):
  File "/usr/lib64/python2.7/threading.py", line 811, in __bootstrap_inner
    self.run()
  File "/usr/lib64/python2.7/threading.py", line 764, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/usr/lib64/python2.7/multiprocessing/pool.py", line 376, in _handle_results
    task = get()
TypeError: ('__init__() takes at least 3 arguments (1 given)', <class 'subprocess32.TimeoutExpired'>, ())
`
code :+1: 
`    try:
        mac_add = subprocess32.check_output(command,timeout=20)
 
    except subprocess32.TimeoutExpired as e:
        mac_add = e`
